### DISCUSSION TITLE
pop-up to review recent games may display games for wrong snake

### DISCUSSION THREAD

**Discussion by bcambl** [BattlesnakeOfficial/feedback#226] [Bug Reports]
Steps to reproduce this _bug_:

1. view a leaderboard ie: _standard_
2. click on a snake to view recent games popup
3. close pop-up
4. click snake on a different snake to view recent games
5. close pop-up
6. click original snake to view recent games and recent games list may be for previously viewed snake

**Comment by originalwebgurl** [comment]
Thanks for letting us know. I'm investigating this today. I'll keep you posted.

**Comment by originalwebgurl** [comment]
Hey @bcambl I just pushed a fix for this (v2.0.22). 

The issue was caused by the modal event listeners not getting cleaned up, the effect was that subsequent calls to show a snake would request data for the current snake selection and all previously selected snakes.
Now that I've cleaned up those event handlers only one request for the selected snake is being sent. 

There can be only one.

Let me know if you're still seeing the issue or if it's working as expected. 

Thanks again for the detailed bug report. It is much appreciated.

**Comment by bcambl** [comment]
Working as expected. Thanks! :tada:
